sql-plugin/src/main/spark400/scala/com/nvidia/spark/rapids/shims/SparkShims.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*** spark-rapids-shim-json-lines
+{"spark": "400"}
+spark-rapids-shim-json-lines ***/
+package com.nvidia.spark.rapids.shims
+
+import com.nvidia.spark.rapids._
+import com.nvidia.spark.rapids.shims.Spark350PlusNonDBShims
+
+import org.apache.spark.sql.catalyst.expressions.{Expression, Literal}
+import org.apache.spark.sql.catalyst.expressions.objects.Invoke
+import org.apache.spark.sql.rapids.shims.InvokeExprMeta
+
+object SparkShimImpl extends Spark350PlusNonDBShims {
+  override def getExprs: Map[Class[_ <: Expression], ExprRule[_ <: Expression]] = {
+    val shimExprs: Map[Class[_ <: Expression], ExprRule[_ <: Expression]] = Seq(
+      GpuOverrides.expr[Literal](
+        "Holds a static value from the query",
+        ExprChecks.projectAndAst(
+          TypeSig.astTypes,
+          (TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL_128 + TypeSig.CALENDAR
+            + TypeSig.BINARY + TypeSig.ARRAY + TypeSig.MAP + TypeSig.STRUCT
+            + TypeSig.ansiIntervals + TypeSig.OBJECT)
```

Comment:
This shim for Literal is not needed if we override the `childExprs` of  `InvokeExprMeta` as below.
```
override val childExprs: Seq[BaseExprMeta[_]] = invoke.arguments.map(GpuOverrides.wrapExpr(_, conf, Some(this)))
```
I post this suggestion is because ObjectType for literal is only for the Invoke, and we can handle it completely in the InvokeExprMeta. GPU does not support object type literal, so we do not want to expose the object type support for literal.